### PR TITLE
[web] add REST APIs third parties (DON'T REVIEW)

### PR DESCRIPTION
**Pull request by tttttangTH**
a draft version for pre-review by @wgtdkp  and @simonlingoogle. Third party library is included.

### CODE REVIEW

File: src/rest/parser.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+class Parser
+{
+public:
+    Parser(Request *aRequest);
+    void Init();
+    void Process(const char *aBuf, int aLength);
+
+private:
+    std::shared_ptr<void> mParser;
```
Comment (tttttangTH) [inline src/rest/parser.hpp:52]: Is this a good practice? @wgtdkp

Comment (wgtdkp) [reply src/rest/parser.hpp:52]: obviously not. Why is this needed?

Comment (wgtdkp) [reply src/rest/parser.hpp:52]: This PR should equals https://github.com/openthread/ot-br-posix/pull/519 + third parties. Any comments or question should happen in https://github.com/openthread/ot-br-posix/pull/519, we simply don't review this PR, but for you to make sure GitHub action tests can success.

**Comment by tttttangTH** [comment]
@wgtdkp @simonlingoogle 

Compared to #519, this PR have all three third party libraries included and it is created for checking whether OTBR-REST could pass Github Action test. Now these three libraries have been added within #519, so we no longer need it. It's time to close it.

Thanks for your review!

**Pull request closed without merging**